### PR TITLE
Make the docker plugin work out-of-the-box with WebappPlugin and Node

### PR DESCRIPTION
**Pull request by jkinkead**
FYI @codeviking , @mjlangan .

### CODE REVIEW

File: src/main/scala/org/allenai/plugins/archetypes/WebappPlugin.scala [hunk 1]
```diff
@@ -1,38 +1,67 @@
...
+      DockerBuildPlugin.autoImport.dockerMainStage.dependsOn(
+        DeployPlugin.autoImport.deployNpmBuild
+      ).value
+    },
+    // Map the node target directory into the docker image.
+    DockerBuildPlugin.autoImport.dockerCopyMappings ++=
+      MappingsHelper.directory(nodeTargetDef.value),
+    // Make `dockerRun` execute node watch, and `dockerStop` stop the watch.
+    DockerBuildPlugin.autoImport.dockerRun :=
+      DockerBuildPlugin.autoImport.dockerRun.dependsOn(NodeKeys.nwatch.in(Npm)).value,
```
Comment (ColinArenz) [inline src/main/scala/org/allenai/plugins/archetypes/WebappPlugin.scala]: I'm a bit concerned here about a rogue watcher. What happens if SBT is killed/interrupted or the container is stopped via another shell process?  Will the watch still get stopped at some point?  If not, does the next `dockerRun` overwrite it?

File: src/main/scala/org/allenai/plugins/archetypes/WebappPlugin.scala [hunk 2]
```diff
@@ -1,38 +1,67 @@
...
+    DockerBuildPlugin.autoImport.dockerStop :=
+      DockerBuildPlugin.autoImport.dockerStop.dependsOn(NodeKeys.unwatch.in(Npm)).value,
+    // When running with `dockerRun`, map the local filesystem's target directory in directly so
+    // that rebuilds don't need to happen to see client-side changes.
+    DockerBuildPlugin.autoImport.dockerRunFlags ++= {
+      val volumeMapping = {
+        val nodeDir = nodeTargetDef.value
+        val imageTarget = DockerBuildPlugin.autoImport.dockerWorkdir.value + '/' + nodeDir.getName
+        s"$nodeDir:$imageTarget"
```
Comment (ColinArenz) [inline src/main/scala/org/allenai/plugins/archetypes/WebappPlugin.scala]: Add the `:readonly` (or `:ro`) modifier, assuming the container doesn't need to modify the JS source.

**Comment by codeviking** [comment]
I don't think I have anything to add, as it's safe to say we won't be using this on S2.  LMK if that's concerning (I'm not sure if you intended for us to eventually port to this)?

**Comment by jkinkead** [comment]
I'll follow-up offline, @codeviking .

File: src/main/scala/org/allenai/plugins/archetypes/WebappPlugin.scala (hunk 2 above)
Comment (jkinkead) [inline src/main/scala/org/allenai/plugins/archetypes/WebappPlugin.scala]: Done.

**Pull request merged**